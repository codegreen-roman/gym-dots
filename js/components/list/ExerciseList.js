import React from 'react'
import { array, string, func } from 'prop-types'
import glamorous from 'glamorous'

import { ExerciseListRow } from './ExerciseListRow'
import { ExerciseListHeader } from './ExerciseListHeader'

const onOrderChangeClick = () => {} // noop

const _ExerciseList = ({ exercises, title, onOrderChangeClick }) => {

    const Ul = glamorous.ul({
        listStyleType: 'none',
        padding: 0
    })

    return (
        <Ul>
            {exercises.length > 0 ? <ExerciseListHeader title={title} /> : null }
            {exercises.map((exercise, idx) => (
                <ExerciseListRow onRowClick={() => onOrderChangeClick(exercise)} key={idx} exercise={exercise} />
            ))}
        </Ul>
    )
}

_ExerciseList.defaultProps = {
    onOrderChangeClick
}

_ExerciseList.propTypes = {
    exercises: array.isRequired,
    title: string.isRequired,
    onOrderChangeClick: func.isRequired
}

export const ExerciseList = _ExerciseList
