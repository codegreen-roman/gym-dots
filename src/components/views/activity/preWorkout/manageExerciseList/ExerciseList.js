import React from 'react'
import { array, string, func } from 'prop-types'
import {
    exerciseList,
    exerciseListHeader,
    exerciseListHeaderDot,
    exerciseListHeaderText
} from './ExerciseList.glamor'
import { ExerciseListRow } from './ExerciseListRow'
import { compose, not, map, isEmpty } from 'ramda'

export const ExerciseListHeader = ({title}) => {
    return (
        <div data-test='list-title' {...exerciseListHeader}>
            <div {...exerciseListHeaderDot} />
            <div {...exerciseListHeaderText}>{title}</div>
        </div>
    )
}

ExerciseListHeader.propTypes = {
    title: string.isRequired
}

const onOrderChangeClick = () => {} // noop

const _ExerciseList = ({ list, title, onOrderChangeClick }) => {
    const row = (item) => (
        <ExerciseListRow key={item.exerciseKey} {...item} onRowClick={() => onOrderChangeClick(item)} />
    )
    const renderRows = () => map(row, list)
    const notEmpty = compose(not, isEmpty)

    return (notEmpty(list) ?
        (<div>
            <ExerciseListHeader title={title} />
            <ul data-test='list' {...exerciseList}>
                {renderRows()}
            </ul>
        </div>) : null
    )
}

_ExerciseList.defaultProps = {
    onOrderChangeClick
}

_ExerciseList.propTypes = {
    list: array.isRequired,
    title: string.isRequired,
    onOrderChangeClick: func
}

export const ExerciseList = _ExerciseList
