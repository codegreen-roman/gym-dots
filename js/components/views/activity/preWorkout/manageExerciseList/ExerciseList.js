import React from 'react'
import { isEmpty } from 'ramda'
import { array, string, func } from 'prop-types'
import { exerciseList, exerciseListHeader } from './ExerciseList.glamor'
import { ExerciseListRow } from './ExerciseListRow'

const onOrderChangeClick = () => {} // noop

const _ExerciseList = ({ list, title, onOrderChangeClick }) => {
    if (isEmpty(list)) {
        return null
    }
    return (
        <div>
            <span data-test='list-title' {...exerciseListHeader}>
                {title}
            </span>
            <ul data-test='list' {...exerciseList}>
                {list.map((itm, idx) =>
                    <ExerciseListRow
                        key={idx}
                        {...itm}
                        onRowClick={() => onOrderChangeClick(itm)}
                    />
                )}
            </ul>
        </div>
    )
}

_ExerciseList.defaultProps = {
    onOrderChangeClick
}

_ExerciseList.propTypes = {
    list: array.isRequired,
    title: string.isRequired,
    onOrderChangeClick: func
}

export const ExerciseList = _ExerciseList
