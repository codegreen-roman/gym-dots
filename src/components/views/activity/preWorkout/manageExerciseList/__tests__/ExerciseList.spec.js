import React from 'react'
import { shallow, mount } from 'enzyme'
import { ExerciseList } from '../ExerciseList'
import R from 'ramda'


const props = {
    title: 'Some title',
    list: [
        {
            exerciseKey: 'xxy',
            restTime: 90,
            name: 'Push-ups',
            sets: 5,
            reps: 20,
            weight: 0,
            results: []
        },
        {
            exerciseKey: 'xyy',
            restTime: 30,
            name: 'Australian pull-ups',
            sets: 5,
            reps: 12,
            weight: 0,
            results: []
        }
    ]
}

const firstExerciseKey = R.compose(R.prop('exerciseKey'), R.head, R.take(1), R.prop('list'))(props)

const setup = props => {
    const actions = {
        onOrderChangeClick: jest.fn()
    }

    const component = shallow(<ExerciseList {...props} {...actions} />)
    const wrapper = mount(<ExerciseList {...props} {...actions} />)

    return {
        firstExerciseKey,
        component,
        wrapper,
        actions,
        props,
        title: component.find('[data-test="list-header"]'),
        list: component.find('[data-test="list"]'),
        row: component.find(`[data-test="exercise-${firstExerciseKey}"]`)
    }
}

describe('ExerciseList component', () => {

    it('should render without crashing', () => {
        const { component } = setup(props)
        expect(component).toMatchSnapshot()
    })

    it('should have list of type ul', () => {
        const { list } = setup(props)
        expect(list.getElement().type).toBe('ul')
    })

    it('should have 2 lis', () => {
        const { component } = setup(props)
        expect(component.find('li').length).toBe(2)
    })

    it('should have correct data-test attribute', () => {
        const { wrapper, firstExerciseKey } = setup(props)
        const firstRow = wrapper.find(`[data-test="exercise-row-${firstExerciseKey}"]`)
        expect(firstRow.props()['data-test']).toBe(`exercise-row-${firstExerciseKey}`)
    })

    it('should have list header', () => {
        const { title } = setup(props)
        expect(title).toMatchSnapshot()
    })

    it('should have list header with title matching text of prop title', () => {
        const { title } = setup(props)
        expect(title.childAt(1).text()).toMatch(/^Some title/)
    })

    describe('When list is empty and has length 0', () => {
        let props
        beforeEach(() => {
            props = {
                onOrderChangeClick: jest.fn(),
                title: '',
                list: []
            }
        })

        it('should not render the component and return null', () => {
            const { wrapper } = setup(props)
            expect(wrapper.find('[data-test="list"]').length).toBe(0)
        })

        it('should have length of rows to be 0', () => {
            const { row } = setup(props)
            expect(row.length).toBe(0)
        })
    })

    describe('Clicking on a row', () => {

        it('should match the snapshot', () => {
            const { wrapper } = setup(props)
            expect(wrapper).toMatchSnapshot()
        })

        it('should call onOrderChangeClick for every row - click', () => {
            const { wrapper, actions: { onOrderChangeClick } } = setup(props)
            const rows = wrapper.find('li')

            for (let r = 0; r < rows.length; r++) {
                rows.at(r).simulate('click')
            }

            expect(onOrderChangeClick).toHaveBeenCalledTimes(rows.length)
        })

        it('should call onOrderChangeClick with list item object', () => {
            const { wrapper, actions: { onOrderChangeClick } } = setup(props)
            const row = wrapper.find('li').at(0)

            row.simulate('click')

            expect(onOrderChangeClick).toHaveBeenCalledWith({
                exerciseKey: 'xxy',
                restTime: 90,
                name: 'Push-ups',
                sets: 5,
                reps: 20,
                weight: 0,
                results: []
            })

        })

    })
})
