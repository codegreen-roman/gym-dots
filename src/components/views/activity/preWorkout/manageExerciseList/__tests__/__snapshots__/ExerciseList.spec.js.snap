// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseList component Clicking on a row should match the snapshot 1`] = `
.css-r04v7w,
[data-css-r04v7w] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  list-style-type: none;
  padding: 0px;
  margin: 0px;
  padding-left: 40px;
  margin-bottom: 20px;
  -webkit-flex-flow: column wrap;
}

.css-1sjmfpn,
[data-css-1sjmfpn] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  border-top: 1px solid #E4EBF5;
  border-bottom: 1px solid #E4EBF5;
  padding: 10px 0 10px 40px;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
}

.css-wjdwya,
[data-css-wjdwya] {
  border-radius: 50%;
  width: 15px;
  height: 15px;
  background: #83D1F3;
  border: 3px solid #46B4E3;
}

.css-iqmdqc,
[data-css-iqmdqc] {
  font-size: 16px;
  font-weight: 500;
  color: #919AA7;
  margin-left: 30px;
  text-transform: capitalize;
}

.css-wd60qt,
[data-css-wd60qt] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
}

.css-eypxw5,
[data-css-eypxw5] {
  width: 100%;
  margin-left: 30px;
  margin-top: 40px;
}

.css-1vhzv1k,
[data-css-1vhzv1k] {
  font-size: 22px;
  font-weight: 600;
  color: #5E6B7D;
  text-transform: capitalize;
}

.css-1q1fiqx,
[data-css-1q1fiqx] {
  font-size: 14px;
  font-weight: 100;
  color: #919AA7;
}

.css-12buxqa,
[data-css-12buxqa] {
  margin-top: -10px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #fff;
  -webkit-flex-flow: column wrap;
  -webkit-align-items: center;
}

.css-ikdvzh,
[data-css-ikdvzh] {
  height: 55px;
  width: 3px;
  background: #83D1F3;
  border-radius: 3px;
}

.css-1fp4adm,
[data-css-1fp4adm] {
  border-radius: 50%;
  width: 15px;
  height: 15px;
  border: 3px solid #46B4E3;
  margin: 12px 0;
}

<Component
  list={
    Array [
      Object {
        "exerciseKey": "xxy",
        "name": "Push-ups",
        "reps": 20,
        "restTime": 90,
        "results": Array [],
        "sets": 5,
        "weight": 0,
      },
      Object {
        "exerciseKey": "xyy",
        "name": "Australian pull-ups",
        "reps": 12,
        "restTime": 30,
        "results": Array [],
        "sets": 5,
        "weight": 0,
      },
    ]
  }
  onOrderChangeClick={[Function]}
  title="Some title"
>
  <div>
    <div
      data-css-1sjmfpn=""
      data-test="list-header"
    >
      <div
        data-css-wjdwya=""
      />
      <div
        data-css-iqmdqc=""
      >
        Some title
      </div>
    </div>
     
    <ul
      data-css-r04v7w=""
      data-test="list"
    >
      <li
        data-css-wd60qt=""
        data-test="exercise-row-xxy"
        key="xxy"
        onClick={[Function]}
      >
        <div
          data-css-12buxqa=""
        >
          <div
            data-css-ikdvzh=""
          />
          <div
            data-css-1fp4adm=""
          />
        </div>
        <div
          data-css-eypxw5=""
        >
          <div
            data-css-1vhzv1k=""
          >
            Push-ups
          </div>
          <div
            data-css-1q1fiqx=""
            data-test="exercise-details"
          >
            0
             kg x 
            20
             reps x 
            5
             sets
          </div>
        </div>
      </li>
      <li
        data-css-wd60qt=""
        data-test="exercise-row-xyy"
        key="xyy"
        onClick={[Function]}
      >
        <div
          data-css-12buxqa=""
        >
          <div
            data-css-ikdvzh=""
          />
          <div
            data-css-1fp4adm=""
          />
        </div>
        <div
          data-css-eypxw5=""
        >
          <div
            data-css-1vhzv1k=""
          >
            Australian pull-ups
          </div>
          <div
            data-css-1q1fiqx=""
            data-test="exercise-details"
          >
            0
             kg x 
            12
             reps x 
            5
             sets
          </div>
        </div>
      </li>
    </ul>
  </div>
</Component>
`;

exports[`ExerciseList component should have list header 1`] = `
.css-1sjmfpn,
[data-css-1sjmfpn] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  border-top: 1px solid #E4EBF5;
  border-bottom: 1px solid #E4EBF5;
  padding: 10px 0 10px 40px;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
}

.css-wjdwya,
[data-css-wjdwya] {
  border-radius: 50%;
  width: 15px;
  height: 15px;
  background: #83D1F3;
  border: 3px solid #46B4E3;
}

.css-iqmdqc,
[data-css-iqmdqc] {
  font-size: 16px;
  font-weight: 500;
  color: #919AA7;
  margin-left: 30px;
  text-transform: capitalize;
}

<div
  data-css-1sjmfpn=""
  data-test="list-header"
>
  <div
    data-css-wjdwya=""
  />
  <div
    data-css-iqmdqc=""
  >
    Some title
  </div>
</div>
`;

exports[`ExerciseList component should render without crashing 1`] = `
.css-r04v7w,
[data-css-r04v7w] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  list-style-type: none;
  padding: 0px;
  margin: 0px;
  padding-left: 40px;
  margin-bottom: 20px;
  -webkit-flex-flow: column wrap;
}

.css-1sjmfpn,
[data-css-1sjmfpn] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  border-top: 1px solid #E4EBF5;
  border-bottom: 1px solid #E4EBF5;
  padding: 10px 0 10px 40px;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-items: center;
}

.css-wjdwya,
[data-css-wjdwya] {
  border-radius: 50%;
  width: 15px;
  height: 15px;
  background: #83D1F3;
  border: 3px solid #46B4E3;
}

.css-iqmdqc,
[data-css-iqmdqc] {
  font-size: 16px;
  font-weight: 500;
  color: #919AA7;
  margin-left: 30px;
  text-transform: capitalize;
}

.css-wd60qt,
[data-css-wd60qt] {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: horizontal;
  -webkit-box-direction: normal;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
}

.css-eypxw5,
[data-css-eypxw5] {
  width: 100%;
  margin-left: 30px;
  margin-top: 40px;
}

.css-1vhzv1k,
[data-css-1vhzv1k] {
  font-size: 22px;
  font-weight: 600;
  color: #5E6B7D;
  text-transform: capitalize;
}

.css-1q1fiqx,
[data-css-1q1fiqx] {
  font-size: 14px;
  font-weight: 100;
  color: #919AA7;
}

.css-12buxqa,
[data-css-12buxqa] {
  margin-top: -10px;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-flow: column wrap;
  flex-flow: column wrap;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #fff;
  -webkit-flex-flow: column wrap;
  -webkit-align-items: center;
}

.css-ikdvzh,
[data-css-ikdvzh] {
  height: 55px;
  width: 3px;
  background: #83D1F3;
  border-radius: 3px;
}

.css-1fp4adm,
[data-css-1fp4adm] {
  border-radius: 50%;
  width: 15px;
  height: 15px;
  border: 3px solid #46B4E3;
  margin: 12px 0;
}

<div>
  <div
    data-css-1sjmfpn=""
    data-test="list-header"
  >
    <div
      data-css-wjdwya=""
    />
    <div
      data-css-iqmdqc=""
    >
      Some title
    </div>
  </div>
   
  <ul
    data-css-r04v7w=""
    data-test="list"
  >
    <li
      data-css-wd60qt=""
      data-test="exercise-row-xxy"
      key="xxy"
      onClick={[Function]}
    >
      <div
        data-css-12buxqa=""
      >
        <div
          data-css-ikdvzh=""
        />
        <div
          data-css-1fp4adm=""
        />
      </div>
      <div
        data-css-eypxw5=""
      >
        <div
          data-css-1vhzv1k=""
        >
          Push-ups
        </div>
        <div
          data-css-1q1fiqx=""
          data-test="exercise-details"
        >
          0
           kg x 
          20
           reps x 
          5
           sets
        </div>
      </div>
    </li>
    <li
      data-css-wd60qt=""
      data-test="exercise-row-xyy"
      key="xyy"
      onClick={[Function]}
    >
      <div
        data-css-12buxqa=""
      >
        <div
          data-css-ikdvzh=""
        />
        <div
          data-css-1fp4adm=""
        />
      </div>
      <div
        data-css-eypxw5=""
      >
        <div
          data-css-1vhzv1k=""
        >
          Australian pull-ups
        </div>
        <div
          data-css-1q1fiqx=""
          data-test="exercise-details"
        >
          0
           kg x 
          12
           reps x 
          5
           sets
        </div>
      </div>
    </li>
  </ul>
</div>
`;
